docs/_docs/03-ug-mutable-dataset.md

```diff
@@ -21,6 +21,19 @@ Now, we offer several options to handle above scenario more efficiently.
 
 ## Options of using index when your dataset changes
 
+### Lineage
+Hyperspace uses lineage for tracing index entries back to the source data files from which they were generated.
+Lineage is required for removing deleted index entries during [index refresh in the incremental mode](#refresh-index---incremental-mode)
+or enforcing deletes at the query time when using [Hybrid Scan](#hybrid-scan).
+By default, lineage is disabled for indexes, and if required it should be enabled at the time of index creation.
+Check the [configuration](https://microsoft.github.io/hyperspace/docs/ug-configuration/) page to see how you can enable lineage for an index.
+
+Once lineage is enabled for an index, Hyperspace adds a new column to the index schema to save the source data file path for each
+index entry. This means enabling lineage increases total storage size used for index files, proportional to the
+number of distinct source data files the index is built on. If you know you wont delete any source data file after index
```

Comment:
```suggestion
number of distinct source data files the index is built on. If you know you will not delete any source file after index
```

Reply:
"source data file" is a term that we are have been using consistently throughout coding, comments and documentation to refer to an index's original dataset files. I would rather keep "source data file" as is.

Reply:
I made this comment because of https://github.com/microsoft/hyperspace/pull/218#discussion_r509664352

I understand the point - "source data files" is also consistent in our code base and github issues, but
just wanted to be consistent in **user doc** and I think it will look better. WDYT? @pirz

Reply:
> Let's define two terms clearly and be consistent throughout user documents :)
> 
> - index source file: source files in source relations
> 
> - index data file: index data file that we built

@pirz @imback82 @apoorvedave1 
Ok let's decide "source data files" or "source files" in user doc. I prefer the latter one because it's shorter and clear. But "source data file" would be good as we're using it in the code base and github issue documentation.